src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -17,102 +29,156 @@ public class ChemoreceptorUpgradeGUI : VBoxContainer, IOrganelleUpgrader
     public NodePath ColourPath = null!;
 
 #pragma warning disable CA2213
+    private OptionButton targetTypes = null!;
     private OptionButton compounds = null!;
+    private Label compoundLabel = null!;
+    private OptionButton species = null!;
+    private Label speciesLabel = null!;
     private Slider maximumDistance = null!;
     private Slider minimumAmount = null!;
     private TweakedColourPicker colour = null!;
 #pragma warning restore CA2213
 
-    private List<Compound>? shownChoices;
+    private List<Compound>? shownCompoundChoices;
+    private List<Species>? shownSpeciesChoices;
 
     public override void _Ready()
     {
+        targetTypes = GetNode<OptionButton>(TargetTypesPath);
         compounds = GetNode<OptionButton>(CompoundsPath);
+        compoundLabel = GetNode<Label>(CompoundsLabelPath);
+        species = GetNode<OptionButton>(SpeciesPath);
+        speciesLabel = GetNode<Label>(SpeciesLabelPath);
         maximumDistance = GetNode<Slider>(MaximumDistancePath);
         minimumAmount = GetNode<Slider>(MinimumAmountPath);
         colour = GetNode<TweakedColourPicker>(ColourPath);
 
         compounds.Clear();
+        species.Clear();
 
         maximumDistance.MinValue = Constants.CHEMORECEPTOR_RANGE_MIN;
         maximumDistance.MaxValue = Constants.CHEMORECEPTOR_RANGE_MAX;
-
         minimumAmount.MinValue = Constants.CHEMORECEPTOR_AMOUNT_MIN;
         minimumAmount.MaxValue = Constants.CHEMORECEPTOR_AMOUNT_MAX;
+
+        TypeChanged(0);
     }
 
-    public void OnStartFor(OrganelleTemplate organelle)
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
     {
-        shownChoices = SimulationParameters.Instance.GetCloudCompounds();
+        // Populate menus
+        targetTypes.AddItem(TranslationServer.Translate("COMPOUNDS"));
+        targetTypes.AddItem(TranslationServer.Translate("SPECIES"));
+
+        shownCompoundChoices = SimulationParameters.Instance.GetCloudCompounds();
+        shownCompoundChoices.Sort((x, y) => string.Compare(x.Name, y.Name, StringComparison.OrdinalIgnoreCase));
 
-        foreach (var choice in shownChoices)
+        foreach (var choice in shownCompoundChoices)
         {
             compounds.AddItem(choice.Name);
         }
 
-        // Select glucose by default
-        var defaultCompoundIndex =
-            shownChoices.FindIndex(c => c.InternalName == Constants.CHEMORECEPTOR_DEFAULT_COMPOUND_NAME);
+        shownSpeciesChoices = currentGame.GameWorld.Map.FindAllSpeciesWithPopulation();
+        shownSpeciesChoices.Sort((x, y) => string.Compare(x.Genus, y.Genus, StringComparison.OrdinalIgnoreCase));
```

Comment:
Why not compare against formatted name? That way the second part of the name would also help in making sure things are alphabetical.